### PR TITLE
Enhance datasource lpstat

### PR DESCRIPTION
**Pull request by wushiqinlou**
Signed-off-by: jiazhang <jiazhang@redhat.com>

### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [x] Is this PR an enhancement?

### Complete Description of Additions/Changes:
Update datasource lpstat to handle more kinds of output

### CODE REVIEW

File: insights/specs/datasources/lpstat.py [hunk 1]
```diff
@@ -37,7 +37,12 @@ def lpstat_protocol_printers_info(broker):
         for line in content:
             if "device for " in line:
                 "Remove printer address information"
-                result.append(line.rsplit(":", 1)[0])
+                [device, address] = line.split(":", 1)
```
Comment (xiangce) [inline insights/specs/datasources/lpstat.py:40]: How about
```
result.append(line.rsplit(":", 1)[0] if '://' in line else line)
```
?

Comment (wushiqinlou) [reply insights/specs/datasources/lpstat.py:40]: @xiangce - For line "device for test_printer3: socket://192.168.1.5:9100", the result of "result.append(line.rsplit(":", 1)[0] if '://' in line else line)" is "device for test_printer3: socket://192.168.1.5", the address will be returned.

Update to use this: `":".join(line.split(":", 2)[0: 2]) if '://' in line else line`

Comment (wushiqinlou) [reply insights/specs/datasources/lpstat.py:40]: @xiangce - For line "device for test_printer3: socket://192.168.1.5:9100", the result of "result.append(line.rsplit(":", 1)[0] if '://' in line else line)" is "device for test_printer3: socket://192.168.1.5", the address will be returned.

Update to use this: `":".join(line.split(":", 2)[0: 2]) if '://' in line else line`

Comment (wushiqinlou) [reply insights/specs/datasources/lpstat.py:40]: @xiangce - For line "device for test_printer3: socket://192.168.1.5:9100", the result of "result.append(line.rsplit(":", 1)[0] if '://' in line else line)" is "device for test_printer3: socket://192.168.1.5", the address will be returned.

Update to use this: `":".join(line.split(":", 2)[0: 2]) if '://' in line else line`

**Review by bfahr — APPROVED** [review]

File: insights/specs/datasources/lpstat.py [hunk 2]
```diff
@@ -37,7 +37,7 @@ def lpstat_protocol_printers_info(broker):
         for line in content:
             if "device for " in line:
                 "Remove printer address information"
-                result.append(line.rsplit(":", 1)[0])
+                result.append(":".join(line.split(":", 2)[0: 2]) if '://' in line else line)
```
Comment (vdymna) [inline insights/specs/datasources/lpstat.py:40]: @wushiqinlou looking at this line for me it is hard to understand what exactly is happening, purely form readability standpoint. What do you think about logic like this?
```python
result.append(line.split("://", 1)[0] if '://' in line else line)
```

Comment (wushiqinlou) [reply insights/specs/datasources/lpstat.py]: @vdymna - Updated, please have a check.

Comment (vdymna) [reply insights/specs/datasources/lpstat.py]: Looks good to me, @xiangce are you good with this approach?

Comment (xiangce) [reply insights/specs/datasources/lpstat.py]: @vdymna - Good to me, thanks!

**Comment by bfahr** [comment]
Fixes and issue parsing the device info in the datasource.

**Pull request merged**